src/main/java/org/mule/extension/aggregator/internal/privileged/executor/SingleGroupAggregatorExecutor.java

```diff
@@ -83,12 +83,14 @@ void registerAsyncAggregationIfNeeded(int delay, TimeUnit unit) {
 
   @Override
   void doScheduleRegisteredAsyncAggregations() {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Refactor this method to reduce its Cognitive Complexity from 16 to the 15 allowed. ([squid:S3776](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS3776))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-core-extensions-mule-aggregators-module&pullRequest=43&issues=AWld7RmvotIXnmsURTyc&open=AWld7RmvotIXnmsURTyc)

[ISSUE_KEY]: AWld7RmvotIXnmsURTyc